[frio] Like buttons improvement

xundeenergie opened the pull request:
This is followup from https://github.com/friendica/friendica/pull/12874

Had to do some cleanup in my commits, and the old PR got lost.

The last action was reducing indentation.

Commit:
simplify code, remove debug output

Commit:
show rotator on button

Commit:
improve optical feedback with rotator on buttons

* functional rotator with feedback from ajax-request for activity
    on like, dislike, share and shareMenu button in wide-screen and media-screen

Commit:
make indentation as tabs

Commit:
improve like/share buttons

change the button only if it could send its request successfully to the server.

fixes:
* disrupting a video on liking or sharing
* timeline jumps around somewhere else, when you like or share a
  posting/comment, and you can not find back.

Commit:
reduce indentation

Commit:
add personal README

Commit:
update README

MrPetovan commented on README-jakob.md:
```diff
@@ -0,0 +1,11 @@
+# Anleitung für die Erweiterungen 
```
Please don’t add these explanation in a file you add to the repository. These belong to the PR itself, not the code.

xundeenergie replied:
Ok I remove the README-jakob.md file.

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,108 @@ function htmlToText(htmlString) {
+		case 'announce':
+			thumbsClass = 'fa-retweet';
+	}
+	// remindert to remove the like-rotator from templates
```
Please remove the rotator from the templates right away.

xundeenergie replied:
Done

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,108 @@ function htmlToText(htmlString) {
+				.addClass('fa')
+				.appendTo($('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child' )).show();
+		} else {
+			$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').show()
```
Please use the [`.attr(attributes)`](https://api.jquery.com/attr/#attr-attributes) form to set the `id` attribute along the `src` one and avoid writing literal HTML as much as possible
```suggestion
			$('<img>').attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
```

xundeenergie committed:
Update view/theme/frio/js/theme.js

in this case, the img[id^=waitfor... .show() shows existing image-containers. All of them which exist.

Because there can be more than one. And this filter just shows them.
adding another rotator with your suggestion can lead to 2 or more rotators on a button.

Co-authored-by: Hypolite Petovan <hypolite@mrpetovan.com>

Commit:
remove reminder for deleting rotators in tpl

Commit:
remove rotator from templates

Commit:
make erromessages translateable

Commit:
move forgotten rotator from template

xundeenergie commented:
I did the changes for the errormessages in the 2 files, you named me (src/App/Page.php and view/theme/frio/templates/js_strings.tpl)
But i don't know, how to translate them. The strings.php, messages.po look all a bit redundant. Please give me a hint, how to translate such text.

Commit:
make errormessages modular and better

Commit:
reduce literal html in javascript-code

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -784,8 +784,8 @@ function doActivityItemAction(ident, verb, un) {
 			// append rotator to the shareMenu-button for small media
-			$('<img id="waitfor-' + verb + '-' + ident.toString() + '">')
-				.attr('src', 'images/rotator.gif')
+      $('<img>')
+        .attr('id', 'waitfor-' + verb + '-' + ident.toString())
```
```suggestion
			$('<img>')
				.attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
```

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -784,8 +784,8 @@ function doActivityItemAction(ident, verb, un) {
+      $('<img>')
+        .attr('id', 'waitfor-' + verb + '-' + ident.toString())
 				.addClass('fa')
 				.appendTo($('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child' )).show();
```
Is the `show()` call even necessary?
```suggestion
				.appendTo($('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child' ))
				.show();
```

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -795,7 +795,8 @@ function doActivityItemAction(ident, verb, un) {
+      .attr('id', 'waitfor-' + verb + '-' + ident.toString())
 			.attr('src', 'images/rotator.gif')
 			.addClass('fa')
 			.appendTo($('button[id^=' + verb + '-' + ident.toString() + '] i:first-child')).show();
```
Is the `show()` call necessary?
```suggestion
			.appendTo($('button[id^=' + verb + '-' + ident.toString() + '] i:first-child'))
			.show();
```

xundeenergie replied:
not neccessary. removed it.

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
+	// if verb is announce, then one rotator is added above to the shareMedia-dropdown button
+	if ($('button:not(button.dropdown-toggle) img#waitfor-' + verb + '-' + ident.toString()).length == 0) {
+		$('<img>')
+      .attr('id', 'waitfor-' + verb + '-' + ident.toString())
+			.attr('src', 'images/rotator.gif')
```
```suggestion
			.attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
```

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
+				// like/dislike buttons
+				$('button[id^=' + verb + '-' + ident.toString() + ']' )
+					.removeClass('active')
+					.attr('onclick', 'doActivityItemAction(' + ident +', "' + verb + '",false )').change();
```
What's the `change()` call doing in this case?
```suggestion
					.attr('onclick', 'doActivityItemAction(' + ident +', "' + verb + '",false )')
					.change();
```

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
+				// link in share-menu
+				$('a[id^=' + verb + '-' + ident.toString() + ']' )
+					.removeClass('active')
+					.attr('href', 'javascript:doActivityItemAction(' + ident +', "' + verb + '",false )').change();
```
```suggestion
					.attr('href', 'javascript:doActivityItemAction(' + ident +', "' + verb + '",false )')
					.change();
```

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
+				// like/dislike buttons
+				$('button[id^=' + verb + '-' + ident.toString() + ']' )
+					.addClass('active')
+					.attr('onclick', 'doActivityItemAction(' + ident + ', "' + verb + '", true )').change();
```
```suggestion
					.attr('onclick', 'doActivityItemAction(' + ident + ', "' + verb + '", true )')
					.change();
```

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
+				// link in share-menu
+				$('a[id^=' + verb + '-' + ident.toString() + ']' )
+					.addClass('active')
+					.attr('href', 'javascript:doActivityItemAction(' + ident + ', "' + verb + '", true )').change();
```
```suggestion
					.attr('href', 'javascript:doActivityItemAction(' + ident + ', "' + verb + '", true )')
					.change();
```

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
+			$('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child').addClass('fa-share');
+			$('button[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
+			$('a[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
+			$('a[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
+      $.jGrowl(aActErr[verb] + '<br>(' + aErrType['srvErr'] + ')', {sticky: false, theme: 'info', life: 5000});
```
Duplicated line and indentation.
```suggestion
			$.jGrowl(aActErr[verb] + '<br>(' + aErrType['srvErr'] + ')', {sticky: false, theme: 'info', life: 5000});
```

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
+		}
+	})
+	.error(function(data){
+			/* Server could not be reaches successfully */
+			$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').remove();
+			$('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child').addClass('fa-share');
+			$('button[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
+			$('a[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
+      $.jGrowl(aActErr[verb] + '<br>(' + aErrType['netErr'] + ')', {sticky: false, theme: 'info', life: 5000});
```
```suggestion
		// Server could not be reached successfully
		$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').remove();
		$('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child').addClass('fa-share');
		$('button[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
		$('a[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
		$.jGrowl(aActErr[verb] + '<br>(' + aErrType['netErr'] + ')', {sticky: false, theme: 'info', life: 5000});
```

MrPetovan reviewed (changes requested):
For translation string changes, once you're done with your code changes, you need to run the following script from the Friendica root folder: `bin/run_xgettext.sh`. This will update the relevant translation files and you just have to commit the produced changes.

Commit:
update translation-files

xundeenergie committed:
Update view/theme/frio/js/theme.js

removed it.

Co-authored-by: Hypolite Petovan <hypolite@mrpetovan.com>

Commit:
fix indetation and removed some calls.

xundeenergie commented:
are there any open requests? 
I can not see them... 
Will this PR go to 2023.03-rc too?

tobiasd commented:
Could you please remove the README-jakob.md file

Commit:
remove own readme

xundeenergie commented:
> Could you please remove the README-jakob.md file

done!

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,103 @@ function htmlToText(htmlString) {
+				.addClass('fa')
+				.appendTo($('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child' ));
+		} else {
+			$('<img>').attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
```
This creates an element but never attaches it to the page DOM, is it intentional?

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,103 @@ function htmlToText(htmlString) {
+			.appendTo($('button[id^=' + verb + '-' + ident.toString() + '] i:first-child'));
+	} else {
+		// show existing rotator for activity
+		$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']')
```
This selects an image but never does anything with it, is it intentional?

Commit:
remove unused selector

was there for historical reasons... :/

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,97 @@ function htmlToText(htmlString) {
+		// remove share-symbol, to replace it by rotator
+		$('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child').removeClass('fa-share');
+		$('button[id^=announce-' + ident.toString() + '] i:first-child').removeClass('fa-retweet');
+		// if no wait-rotator for activity(verb) is added, add it. or just show it, if exists
```
```suggestion
		// if no wait-rotator exists for activity(verb), add it.
```

xundeenergie replied:
changed line to
"avoid multiple rotators on like/share-button if klicked multiple times"

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,97 @@ function htmlToText(htmlString) {
+			.addClass('fa')
+			.appendTo($('button[id^=' + verb + '-' + ident.toString() + '] i:first-child'));
+	}
+	// do request for activity
```
Redundant with code
```suggestion

```

MrPetovan commented on view/theme/frio/js/theme.js:
```diff
@@ -764,12 +764,97 @@ function htmlToText(htmlString) {
+	.success(function(data){
+		$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').remove();
+		if (data.status == 'ok') {
+			// response from server was ok
```
Comment redundant with code.
```suggestion
```

MrPetovan reviewed (changes requested):
Last touches.

Commit:
change/remove comments in code

MrPetovan commented:
Thank you for your work!

xundeenergie commented:
> Thank you for your work!

Thank you for your attendation and understanding a newcomer :)

MrPetovan commented:
Closes #12859
